openfl-workspace/keras_nlp/code/dataloader_utils.py

```diff
@@ -0,0 +1,231 @@
+
+    input_characters = sorted(list(input_characters))
+    target_characters = sorted(list(target_characters))
+    num_encoder_tokens = len(input_characters)
+    num_decoder_tokens = len(target_characters)
+    max_encoder_seq_length = max([len(txt) for txt in input_texts])
+    max_decoder_seq_length = max([len(txt) for txt in target_texts])
+
+    details = {"num_samples": len(input_texts),
+               "num_encoder_tokens": num_encoder_tokens,
+               "num_decoder_tokens": num_decoder_tokens,
+               "max_encoder_seq_length": max_encoder_seq_length,
+               "max_decoder_seq_length": max_decoder_seq_length}
+
+    input_token_index = dict([(char, i) for i, char in enumerate(input_characters)])
+    target_token_index = dict([(char, i) for i, char in enumerate(target_characters)])
+
+    encoder_input_data = np.zeros(
+        (len(input_texts), max_encoder_seq_length, num_encoder_tokens), dtype="float32")
+
+    decoder_input_data = np.zeros(
+        (len(input_texts), max_decoder_seq_length, num_decoder_tokens), dtype="float32")
+
+    decoder_target_data = np.zeros(
+        (len(input_texts), max_decoder_seq_length, num_decoder_tokens), dtype="float32")
+
+    for i, (input_text, target_text) in enumerate(zip(input_texts, target_texts)):
+        for t, char in enumerate(input_text):
+            encoder_input_data[i, t, input_token_index[char]] = 1.0
+        encoder_input_data[i, t + 1:, input_token_index[" "]] = 1.0
+        for t, char in enumerate(target_text):
+            # decoder_target_data is ahead of decoder_input_data by one timestep
+            decoder_input_data[i, t, target_token_index[char]] = 1.0
+            if t > 0:
+                # decoder_target_data will be ahead by one timestep
+                # and will not include the start character.
+                decoder_target_data[i, t - 1, target_token_index[char]] = 1.0
+        decoder_input_data[i, t + 1:, target_token_index[" "]] = 1.0
+        decoder_target_data[i, t:, target_token_index[" "]] = 1.0
+
+    logger.info(f"[DL]-import_raw_data: Number of samples = {len(input_texts)}")
+    logger.info(f"[DL]-import_raw_data: Number of unique input tokens = {num_encoder_tokens}")
+    logger.info(f"[DL]-import_raw_data: \
+                  Number of unique decoder tokens = {num_decoder_tokens}")
+
+    logger.info(f"[DL]-import_raw_data: \
+                  Max sequence length for inputs = {max_encoder_seq_length}")
+
+    logger.info(f"[DL]-import_raw_data: \
+                  Max sequence length for outputs = {max_decoder_seq_length}")
+
+    logger.info(f"[DL]-import_raw_data: encoder_input_data = {encoder_input_data.shape}")
+    logger.info(f"[DL]-import_raw_data: decoder_input_data = {decoder_input_data.shape}")
+    logger.info(f"[DL]-import_raw_data: decoder_target_data = {decoder_target_data.shape}")
+
+    return details, encoder_input_data, decoder_input_data, decoder_target_data
+
+
+<<<<<<< HEAD
```

Comment:
It looks like there may have been some conflicts from your local merge here. This should be cleaned up